tests/mocks/CMakeLists.txt

```diff
@@ -14,7 +14,7 @@
 # License along with this library; if not, write to the Free Software
 # Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
 
-set( CMAKE_CXX_STANDARD 14 )
+set( CMAKE_CXX_STANDARD 17 )
```

Comment:
I think we should add set( CMAKE_CXX_STANDARD 17 ) & set( CMAKE_CXX_STANDARD_REQUIRED ON ) in the parent CMakeLists.txt too.

Otherwise we only compile 7 for tests.